### PR TITLE
Proof transformations cleanup - part 2 of N: Refactor ProofGraph::mergeClauses

### PR DESCRIPTION
**Pull request by blishko**
The implementation of the method has been simplified and assertions for preconditions and postconditions have been added.
Several unit tests have been added to check correctness of the implementation.

### CODE REVIEW

**Comment by aehyvari** [comment]
I added a suggestion that would avoid a bit the duplication of the code.  Do you think it'd be better this way or do you prefer the old version?

File: src/proof/PGHelp.cc [hunk 1]
```diff
@@ -220,37 +220,35 @@ bool ProofGraph::mergeClauses(std::vector<Lit> const & A, std::vector<Lit> const
...
-            if (var(A[i]) != pivot) {
-                assert(res == 0 or resolv[res - 1] != A[i]); // no duplicates
-                resolv[res++] = A[i];
-            }
+        if (A[i] == B[j]) {
+            addIfNotPivot(A[i]);
+            ++i;
+            ++j;
+        } else if (A[i] < B[j]) {
+            addIfNotPivot(A[i]);
```
Comment (blishko) [inline src/proof/PGHelp.cc:236]: I like the lambda, but here I prefer my version of the `if` structure, so we do not need to call the lambda in both branches.
```
if (A[i] <= B[j] {
	if (A[i] == B[j] { ++j };
	addIfNotPivot(A[i]);
	++i;
}
```

Comment (aehyvari) [reply src/proof/PGHelp.cc:236]: Ok, let's go with that then if you prefer!

**Review by aehyvari — APPROVED** [review]

**Comment by blishko** [comment]
Looks like I need to rebase. I am curious what will happen to the other PRs. In theory, after merging this one, the next one should be automatically redirected to merge to master instead of this one. Let's see if it works that way.

**Pull request merged**